src/file_handler.rs

```diff
@@ -123,22 +124,126 @@ fn config_dir_path() -> String {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use file_handler::ConfigManagement;
+    use file_handler::FileManagement;
     use file_handler::FileSystem;
+    use std::fs;
     use std::io;
 
     struct MockFileSystem;
+    struct MockFileHandler;
 
     impl FileSystem for MockFileSystem {
         fn create_dir(&self, _path: &str) -> io::Result<()> {
-            Ok(())
+            fs::create_dir_all(&_path)
```

Comment:
The reason I chose to have this mock implementation is because I might want to test this function being called inside of other functions, and I don't actually want to test the `fs::create_dir_all` implementation - I expect that to be tested in the `fs` module :) With this implementation we're _actually_ creating a dir on the file system which is a side-effect that we should avoid. What if the disk happens to be full? Or we're trying to create it in a place where we don't have permission to? The test will fail but _our code_ wasn't the problem.

I hope I'm making sense! I saw your comment about this but I still don't understand why we need this.